src/utils/memoFn.ts

```diff
@@ -0,0 +1,79 @@
+// let cachedCalls = 0;
+
+import logger from './logger';
+
+/**
+ * Memoization wrapper for common used PURE helper functions.
+ * The function will be cached only if it receives number, string or boolean as params
+ * PURE means that function accepts at least a single argument and returns
+ * same result all the time same arguments are used.
+ * In order to memoize the function, it should be defined only once as a static function - outside react components.
+ *
+ *
+ * @export
+ * @param {Function} fn - pure function with at least single argument
+ * @returns {Function} - same function with a cache
+ */
+export function memoFn<TArgs extends unknown[], TReturn extends unknown>(
```

Comment:
Should this be `TArgs extends Array<string | number | boolean>`?

Reply:
Yeah, I tried it, didn't work:
https://www.typescriptlang.org/play?#code/MYewdgzgLgBARgVwJYBsAmAlA5nAhjAXhgFsBTYkAMTAAoAoGGG0FEAJwC4Zo2kwsANDFwoADgAtcXMAmJxSbQjACMASkIA+GAAM2OXDQAkAbxbsAvgJMiJuc6u11VAbjp1SAD1HtYAMwRgwFBI4CTkVGAAPAAqAIJ6EDCeUKRgaInxbLgAnpE8fFgwAD7wICAopLhgxTAycgoaQtEYpFAIbNXJqekwAQDWYCAA7mAaNL5gXDQAdLO4CVxxCeoEWs2t7WBOXFDZoqQgvjATMMYMjDBsGx3HYK7mQA

Reply:
I mean we should infer the arguments of the function we wrap so in the end it's still the same function.
Would be good to figure it out later though